src/cargo/util/toml/mod.rs

```diff
@@ -686,6 +688,40 @@ fn default_readme_from_package_root(package_root: &Path) -> Option<String> {
     None
 }
 
+#[tracing::instrument(skip_all)]
+fn resolve_features(
+    original_features: Option<&BTreeMap<manifest::FeatureName, Vec<String>>>,
+    edition: Edition,
+) -> CargoResult<Option<BTreeMap<manifest::FeatureName, Vec<String>>>> {
+    let Some(mut resolved_features) = original_features.cloned() else {
+        return Ok(None);
+    };
+
+    if Edition::Edition2024 <= edition {
+        for activations in resolved_features.values_mut() {
+            let mut deps = Vec::new();
+            for feature_value in activations.iter() {
+                let feature_value = FeatureValue::new(InternedString::new(feature_value));
```

Comment:
So, this PR chose the approach:

> Implicitly create `dep:serde` if `serde/derive` is used

I forgot if we had a conclusion somewhere.

Reply:
See https://github.com/rust-lang/cargo/issues/14016#issuecomment-2161083450

> We discussed this in today's team meeting and lean towards `"dep_name/feature_name"` implying `"dep:dep_name"` in Edition 2024 with it being normalized in the published `Cargo.toml` file.

Reply:
Why did we choose to do it here instead of in `Summary`?

https://github.com/rust-lang/cargo/blob/471287ffed354a468114caa5c72db97da6b6ddfb/src/cargo/util/toml/mod.rs#L1442-L1459

Reply:
We want everything to be normalized on `resolved_toml`, right?

Reply:
By doing it here, we get `cargo publish` normalization "for free" and all data-structures are consistent for less chance of bugs (which also makes it easier to introspect for debugging)

Reply:
> Implicitly create `dep:serde` if `serde/derive` is used

Doesn't this fail if `serde` is not an optional dependency?